### PR TITLE
swap fileset for glob, Fixes #638

### LINKED ISSUE #638: Deprecate fileset
Hi!

I'm opening this issue to start a thread on the possibility to deprecate https://github.com/mklabs/node-fileset

I recently merged a pending PR from @isaacs (mklabs/node-fileset#22), which

> may or may not be a breaking change from fileset's pov. ([changelog](https://github.com/mklabs/node-fileset/blob/master/CHANGELOG.md#100-2016-06-01))

to quote Isaacs. So I bumped filset to a major 1.0 version which should save you from any trouble.

---

I really really appreciate that you choose to rely on it and build istanbul on top of it. But, to be honest, fileset is a package I wrote a long time ago and never really used it. I always either used just glob with the ignore pattern or redone the filtering on top of globs results before glob 5.

I'm considering deprecating the package in favour of newer, better similar packages like @sindresorhus https://github.com/sindresorhus/globby, or simply glob itself with the `options.ignore` patterns.

This should help with performance (as the minimatch ignore is done internally by glob, avoiding one costly glob call on all ignore patterns) and issues like #556 

I can maybe help in the migration process by opening a PR to switch from fileset to globby, or just glob's with `options.ignore`.

Let me know :)

Thanks for building Istanbul.

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -111,7 +111,7 @@
     "async": "1.x",
     "escodegen": "1.8.x",
     "esprima": "2.7.x",
-    "fileset": "0.2.x",
+    "glob": "^5.0.15",
```
Comment (lo1tuma) [inline package.json]: `glob` is already a `devDependency`, so you should remove it there.

Comment (graingert) [inline package.json]: Ah woops

Comment (graingert) [inline package.json]: This is probably redundant with the nyc merge

File: test/other/test-matcher.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ var path = require('path'),
 module.exports = {
     setUp: function (cb) {
         if (!allFiles) {
-            fileset('**/*.js', '', { cwd: root}, function (err, files) {
+            glob('**/*.js', '', { cwd: root}, function (err, files) {
```
Comment (popomore) [inline test/other/test-matcher.js]: Should delete the second paramter

```diff
- glob('**/*.js', '', { cwd: root}, function (err, files) {
+ glob('**/*.js', { cwd: root}, function (err, files) {
```

Comment (popomore) [inline test/other/test-matcher.js]: Then can make ci happy.

**Comment by maxkoryukov** [comment]
Looks, like there is an overall solution: #673

**Pull request closed without merging**